### PR TITLE
refactor(libraries/tipipeline): refactor global component functions to support hotfix pull requests

### PR DESCRIPTION
**Pull request by wuhuizuo**
## Functions

- `comonent.checkout`
- `comonent.fetchAndExtractArtifact`

> also make the checkout timeout as optional param.

### CODE REVIEW

**Comment by wuhuizuo** [comment]
tested records:
- https://do.pingcap.net/jenkins/job/pingcap/job/tidb/job/release-6.1/job/ghpr_mysql_test/423/
- https://do.pingcap.net/jenkins/job/pingcap/job/tidb/job/release-6.1/job/ghpr_mysql_test/422/

File: libraries/tipipeline/vars/component.groovy [hunk 1]
```diff
@@ -1,26 +1,37 @@
...
+def computeBranchFromPR(String keyInComment, String prTargetBranch, String prCommentBody, String trunkBranch="master") {
     // /run-xxx dep1=release-x.y
-    final commentBodyReg = /\b${keyInComment}\s*=\s*([^\s\\]+)(\s|\\|$)/    
+    final commentBodyReg = /\b${keyInComment}\s*=\s*([^\s\\]+)(\s|\\|$)/
     // - release-6.2
     // - release-6.2-20220801
     // - 6.2.0-pitr-dev    
     final releaseOrHotfixBranchReg = /^(release\-)?(\d+\.\d+)(\.\d+\-.+)?/
+    // - release-6.1-20230101-v6.1.2, the patch number should > 0
+    final newHotfixBranchReg = /^release\-\d+\.\d+\-\d+\-v(\d+\.\d+\.[1-9]\d*)/
```
[the remark this answers is not in the document]
Comment (wuhuizuo) [reply libraries/tipipeline/vars/component.groovy]: reverted

**Review by purelind — APPROVED** [review]
/lgtm

**Review by purelind — APPROVED** [review]
/lgtm

**Pull request merged**